src/apps/backend/modules/workflow/errors.py

```diff
@@ -0,0 +1,56 @@
+from modules.error.custom_errors import AppError
+from modules.workflow.types import WorkflowErrorCode
+
+
+class WorkflowIdNotFoundError(AppError):
+    def __init__(self, workflow_id: str) -> None:
+        super().__init__(
+            code=WorkflowErrorCode.WORKFLOW_WITH_ID_NOT_FOUND,
+            http_status_code=404,
+            message=f"Could not find a workflow with id: {workflow_id}. Please verify and try again.",
```

Comment:
As mentioned earlier, the errors should not only surface problem but also offer solution. Kindly Google "how to write good error messages" and address them at all places in this PR